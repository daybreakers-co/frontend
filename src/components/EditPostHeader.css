.edit.PostHeader input {
  text-align: center;
  background: transparent;
  appearance: none;
  box-shadow: none;
  border-radius: none;
  outline: none;
  border: none
}

.PostHeader.edit .fa {
  font-size: 1.25rem;
  margin-right: 0.5rem;
}

.PostHeader.edit .public-DraftStyleDefault-ltr {
  text-align: center;
}

.PostHeader.edit .public-DraftEditorPlaceholder-root {
  text-align: center;
  width: 100%;
}

.PostHeader.edit .H-Large {
  width: 100%;
}

.PostHeader.edit .T-Large {
  width: 100%;
}

.PostHeader.edit dates {
  z-index: 100;
}
