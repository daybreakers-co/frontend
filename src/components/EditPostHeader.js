import React from 'react'
import PropTypes from 'prop-types'
import Dropzone from 'react-dropzone'
import UploadableImage from './UploadableImage'
import ScaledImage from './ScaledImage'

import DateRangeInput from './DateRangeInput';

import './PostHeader.css'
import './EditPostHeader.css'

class EditPostHeader extends React.Component {
  constructor(props) {
    super()
    this.state = {
      title: props.title,
      subtitle: props.subtitle,
      file: null
    }
  }

  static propTypes = {
    title: PropTypes.string,
    subtitle: PropTypes.string,
    header: PropTypes.object,
    startDate: PropTypes.string,
    endDate: PropTypes.string,
    uploadParentId: PropTypes.string.isRequired,
    uploadParentType: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      title: nextProps.title,
      subtitle: nextProps.subtitle,
      header: nextProps.header
    })
  }

  onDrop = (files) => {
    this.setState({file: files[0]})
  }

  handleBlur = () => {
    this.props.onChange({
      title:    this.state.title,
      subtitle: this.state.subtitle
    })
  }

  render () {
    const { header, uploadParentId, uploadParentType, startDate, endDate } = this.props
    const { title, subtitle, file } = this.state
    let backgroundImage;

    if (file) {
      backgroundImage = <UploadableImage
      parentType={uploadParentType}
      parentId={uploadParentId}
      file={this.state.file}
      cover />
    } else if (header) {
      backgroundImage = <ScaledImage
      key={header.id}
      image={header}
      style={{flex: header.ratio}}
      cover
      alt="image"/>
    }

    return (
      <Dropzone onDrop={this.onDrop} className="PostHeader edit" disableClick={true}>
        {backgroundImage}
        <hgroup>
          <input
            className="H-Large"
            value={title || ""}
            placeholder="Enter the title of your post"
            onChange={(e) => this.setState({title: e.target.value})}
            onBlur={this.handleBlur} />
          <dates>
            <DateRangeInput
              startDate={startDate}
              endDate={endDate}
              onChange={(result) => this.props.onChange(result)}
              />
          </dates>
          <input
            className="T-Large"
            value={subtitle || ""}
            placeholder="Enter an introduction to your post"
            onChange={(e) => this.setState({subtitle: e.target.value})}
            onBlur={this.handleBlur} />
        </hgroup>
      </Dropzone>
    )
  }
}

export default EditPostHeader;
