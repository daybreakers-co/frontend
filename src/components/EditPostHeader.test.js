import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';

import EditPostHeader from './EditPostHeader'

describe("EditPostHeader", () => {
  const photo = {id: "def234", ratio: 1, url: "image.jpg"}
  const file = {}
  const spy = jest.fn()

  describe("With photo", () => {
    it("Renders a scaled image", () => {
      const wrapper = shallow(
        <EditPostHeader
          title="Header title"
          subtitle="Header subtitle"
          uploadParentId="trip1"
          uploadParentType="Trip"
          header={photo}
          onChange={spy} />
      )

      let scaledImage = wrapper.find("ScaledImage")
      expect(scaledImage.prop("image")).toEqual(photo)
      expect(scaledImage.prop("style")).toEqual({flex: 1})
    })
  })

  describe("With file", () => {
    it("Renders an UploadableImage", () => {
      const wrapper = shallow(
        <EditPostHeader
          title="Header title"
          subtitle="Header subtitle"
          uploadParentId="trip1"
          uploadParentType="Trip"
          header={photo}
          onChange={spy} />
      )
      wrapper.setState({file: file})

      let uploadableImage = wrapper.find("UploadableImage")
      expect(uploadableImage.prop("parentId")).toEqual("trip1")
      expect(uploadableImage.prop("parentType")).toEqual("Trip")
      expect(uploadableImage.prop("file")).toEqual(file)
    })
  })

  it("Uploads an image when file is dropped", () => {
    const wrapper = shallow(
      <EditPostHeader
        title="Header title"
        subtitle="Header subtitle"
        uploadParentId="trip1"
        uploadParentType="Trip"
        header={photo}
        onChange={spy} />
    )
    wrapper.instance().onDrop.call(wrapper.instance(), [file])

    expect(wrapper.instance().state.file).toEqual(file)
  })

  it ("Matches snapshot",() => {
    const wrapper = shallow(
      <EditPostHeader
        title="Header title"
        subtitle="Header subtitle"
        uploadParentId="trip1"
        uploadParentType="Trip"
        header={photo}
        onChange={spy} />
    )

    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
