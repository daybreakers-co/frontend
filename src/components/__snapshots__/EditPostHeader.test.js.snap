// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPostHeader Matches snapshot 1`] = `
<Dropzone
  className="PostHeader edit"
  disableClick={true}
  disablePreview={false}
  maxSize={Infinity}
  minSize={0}
  multiple={true}
  onDrop={[Function]}
  preventDropOnDocument={true}
>
  <ScaledImage
    alt="image"
    cover={true}
    image={
      Object {
        "id": "def234",
        "ratio": 1,
        "url": "image.jpg",
      }
    }
    key="def234"
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <hgroup>
    <PlainTextAreaInput
      className="H-Large"
      onBlur={[Function]}
      placeholder="Enter the title of your post"
      value="Header title"
    />
    <dates>
      <DateRangeInput
        onChange={[Function]}
      />
    </dates>
    <PlainTextAreaInput
      className="T-Large"
      onBlur={[Function]}
      placeholder="Enter an introduction to your trip"
      value="Header subtitle"
    />
  </hgroup>
</Dropzone>
`;
